verifiers/internal/gha/provenance.go

```diff
@@ -297,7 +321,25 @@ func VerifyProvenance(env *dsselib.Envelope, provenanceOpts *options.ProvenanceO
 		if err := isValidDelegatorBuilderID(prov); err != nil {
 			return err
 		}
-		// Note: `provenanceOpts.ExpectedBuilderID` is provided by the user.
+
+		// If ExpectedID is empty, check to see if it is a trusted builder.
+		// If empty and trusted builder, populate with path from provenance,
+		// otherwise, populate from user input.
+		//
+		// Example: the expectedBuilderID will default to the delegator builder ID for BYOB.
+		// This can verify actual BYOB builderIDPath against the trusted builderIDPath inputted,
+		// Currently slsa-framework path is the only one supported for ExpectedBuilderPath.
+		if builderOpts.ExpectedID == nil || *builderOpts.ExpectedID == "" {
+			var trustedBuilderRepositoryPath = httpsGithubCom + trustedBuilderRepository + "/.github/workflows/"
+			if provenanceOpts.ExpectedBuilderID, err = verifyBuilderIDPath(prov, trustedBuilderRepositoryPath); err != nil {
+				return err
+			}
+		} else {
+			provenanceOpts.ExpectedBuilderID = *builderOpts.ExpectedID
+		}
+
+		// Note: `provenanceOpts.ExpectedBuilderID` is provided by the user
```

Comment:
nit: NOTE is a kind of label, like "TODO"
```suggestion
		// NOTE: `provenanceOpts.ExpectedBuilderID` is provided by the user
```

Reply:
Got it, thanks.